glmnet/linear.py

```diff
@@ -180,9 +178,12 @@ def fit(self, X, y, sample_weight=None, relative_penalties=None):
         X : array, shape (n_samples, n_features)
             Input features
 
-        Y : array, shape (n_samples,)
+        y : array, shape (n_samples,)
             Target values
 
+        groups: array, shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into train/test set.
```

Comment:
I think the docstring here should specify what happens when `None` is provided (e.g., "If None, then data will be split randomly for K-fold cross-validation.").

Also, I think this description should mention that the groups will be passed to sklearn.model_selection.GroupKFold. Without a link to that, it seems like somebody would need to look through the code to figure out how this `groups` variable is going to work